test/e2e/apps/rc.go

```diff
@@ -315,7 +327,9 @@ var _ = SIGDescribe("ReplicationController", func() {
 				return true, nil
 			})
 			framework.ExpectNoError(err, "Wait until condition with watch events should not return an error")
-			framework.ExpectEqual(eventFound, true, "Failed to find updated ready replica count")
+			if !eventFound {
```

Comment:
is this condition needed at all? We are not setting eventFound in the previous block.

Reply:
It gets initialized as false in line 312 and set to true inside the callback function in line 326, so this check seems legitimate to me.